First will start by importing the libraries that we need Pandas, Numpy, matplotlib & seaborn

In [93]:
import pandas as pd
from pandas import DataFrame, Series
import numpy as np
import matplotlib.pyplot as plt
from pylab import rcParams
import seaborn as sns
import numpy as np
import pandas as pd 
from sklearn.datasets import load_iris
from sklearn.neighbors import KNeighborsClassifier
from sklearn import metrics
from datetime import date
%matplotlib inline

Then we need to read our data

In [24]:
df = pd.read_csv("C:\\Users\\mrnaz\\Desktop\\BootCamp\\My_Project\\WN-JAN-2020.csv")

let's have a look at what just we read

In [55]:
df.shape

(107708, 30)

In [62]:
df.head(5)

MONTH_DAY  WEEK_DAY   FL_DATE  FL_NUM TAIL_NUM  ORIGIN_CITY ORIGIN_STATE  \
0          1         3  1/1/2020    5888   N951WN  Ontario, CA   California   
1          1         3  1/1/2020    6276   N467WN  Ontario, CA   California   
2          1         3  1/1/2020    4598   N7885A  Ontario, CA   California   
3          1         3  1/1/2020    4761   N551WN  Ontario, CA   California   
4          1         3  1/1/2020    5162   N968WN  Ontario, CA   California   

           DEST_CITY  DEST_STATE  CRS_DEP_TIME  ...  ARR_DELAY_NEW  \
0  San Francisco, CA  California          1810  ...           68.0   
1  San Francisco, CA  California          1150  ...            0.0   
2       San Jose, CA  California          2020  ...            0.0   
3       San Jose, CA  California          1340  ...           10.0   
4       San Jose, CA  California           915  ...            0.0   

   CRS_ELAPSED_TIME  ACTUAL_ELAPSED_TIME  AIR_TIME  DISTANCE  CARRIER_DELAY  \
0                95                122.0      74.0       363            8.0   
1                90                 92.0      71.0       363            0.0   
2                70                 68.0      57.0       333            0.0   
3                75                 75.0      63.0       333            0.0   
4                80                 67.0      57.0       333            0.0   

   WEATHER_DELAY  NAS_DELAY  SECURITY_DELAY  LATE_AIRCRAFT_DELAY  
0            0.0       27.0             0.0                 33.0  
1            0.0        0.0             0.0                  0.0  
2            0.0        0.0             0.0                  0.0  
3            0.0        0.0             0.0                  0.0  
4            0.0        0.0             0.0                  0.0  

[5 rows x 30 columns]

First I will rename some columns 

In [26]:
df.rename({'MKT_CARRIER_FL_NUM':'FL_NUM', 
               'ORIGIN_CITY_NAME':'ORIGIN_CITY',
               'ORIGIN_CITY_NAME':'ORIGIN_CITY',
               'ORIGIN_STATE_NM':'ORIGIN_STATE',
               'DEST_CITY_NAME':'DEST_CITY',
              'DAY_OF_MONTH':'MONTH_DAY', 
               'DAY_OF_WEEK':'WEEK_DAY',
               'DEST_STATE_NM':'DEST_STATE'},
              axis=1, inplace=True)

Second I will remove the Time in Hourly Block and the 3 letters city code it is not familiar for most people that are not working in the aviation let's just remove it and save some space

In [27]:
df.drop(['ORIGIN','DEST','DEP_TIME_BLK','ARR_TIME_BLK'],axis=1,inplace=True)

we decided that we will not include canceled flights in our study 
so we will remove the row that have tha value of 1 in the column CANCELLED then we will remove the columns CANCELLED and CANCELLATION_CODE

In [28]:
df = df[df.CANCELLED == 0]

In [29]:
df.drop(['CANCELLED','CANCELLATION_CODE'],axis=1,inplace=True)

let's have a look on the table after we do this changes

In [41]:
df.head(100)# or df.sample(5)

MONTH_DAY  WEEK_DAY   FL_DATE  FL_NUM TAIL_NUM  ORIGIN_CITY ORIGIN_STATE  \
0            1         3  1/1/2020    5888   N951WN  Ontario, CA   California   
1            1         3  1/1/2020    6276   N467WN  Ontario, CA   California   
2            1         3  1/1/2020    4598   N7885A  Ontario, CA   California   
3            1         3  1/1/2020    4761   N551WN  Ontario, CA   California   
4            1         3  1/1/2020    5162   N968WN  Ontario, CA   California   
..         ...       ...       ...     ...      ...          ...          ...   
96           1         3  1/1/2020    3239   N8309C  Phoenix, AZ      Arizona   
97           1         3  1/1/2020    3896   N8621A  Phoenix, AZ      Arizona   
98           1         3  1/1/2020    4833   N707SA  Phoenix, AZ      Arizona   
99           1         3  1/1/2020    5763   N774SW  Phoenix, AZ      Arizona   
100          1         3  1/1/2020    3689   N401WN  Phoenix, AZ      Arizona   

             DEST_CITY  DEST_STATE  CRS_DEP_TIME  ...  ARR_DELAY_NEW  \
0    San Francisco, CA  California          1810  ...           68.0   
1    San Francisco, CA  California          1150  ...            0.0   
2         San Jose, CA  California          2020  ...            0.0   
3         San Jose, CA  California          1340  ...           10.0   
4         San Jose, CA  California           915  ...            0.0   
..                 ...         ...           ...  ...            ...   
96       Nashville, TN   Tennessee           700  ...            0.0   
97       Nashville, TN   Tennessee           935  ...            0.0   
98       Nashville, TN   Tennessee          1930  ...            0.0   
99       Nashville, TN   Tennessee          1520  ...            0.0   
100          Boise, ID       Idaho          1510  ...            0.0   

     CRS_ELAPSED_TIME  ACTUAL_ELAPSED_TIME  AIR_TIME  DISTANCE  CARRIER_DELAY  \
0                  95                122.0      74.0       363            8.0   
1                  90                 92.0      71.0       363            0.0   
2                  70                 68.0      57.0       333            0.0   
3                  75                 75.0      63.0       333            0.0   
4                  80                 67.0      57.0       333            0.0   
..                ...                  ...       ...       ...            ...   
96                190                166.0     154.0      1449            0.0   
97                190                175.0     156.0      1449            0.0   
98                190                168.0     154.0      1449            0.0   
99                185                172.0     157.0      1449            0.0   
100               130                127.0     111.0       735            0.0   

     WEATHER_DELAY  NAS_DELAY  SECURITY_DELAY  LATE_AIRCRAFT_DELAY  
0              0.0       27.0             0.0                 33.0  
1              0.0        0.0             0.0                  0.0  
2              0.0        0.0             0.0                  0.0  
3              0.0        0.0             0.0                  0.0  
4              0.0        0.0             0.0                  0.0  
..             ...        ...             ...                  ...  
96             0.0        0.0             0.0                  0.0  
97             0.0        0.0             0.0                  0.0  
98             0.0        0.0             0.0                  0.0  
99             0.0        0.0             0.0                  0.0  
100            0.0        0.0             0.0                  0.0  

[100 rows x 30 columns]

O.K we need to know more about our table such as number of non-null and data type for each column

In [68]:
df.corr()  # View the correlations

MONTH_DAY  WEEK_DAY    FL_NUM  CRS_DEP_TIME  DEP_TIME  \
MONTH_DAY             1.000000 -0.048515 -0.146855     -0.002550 -0.007260   
WEEK_DAY             -0.048515  1.000000  0.398623      0.004201  0.009068   
FL_NUM               -0.146855  0.398623  1.000000     -0.008850 -0.004664   
CRS_DEP_TIME         -0.002550  0.004201 -0.008850      1.000000  0.994652   
DEP_TIME             -0.007260  0.009068 -0.004664      0.994652  1.000000   
DEP_DELAY            -0.069803  0.078215  0.046779      0.106873  0.170378   
DEP_DELAY_NEW        -0.059601  0.068159  0.039619      0.107251  0.169952   
TAXI_OUT              0.034694  0.008700  0.004223     -0.107235 -0.102996   
WHEELS_OFF           -0.006342  0.008880 -0.005186      0.992801  0.997369   
WHEELS_ON             0.013104  0.000363 -0.010285      0.773774  0.774285   
TAXI_IN              -0.020175  0.025283  0.023904     -0.018353 -0.017733   
CRS_ARR_TIME          0.013377 -0.004497 -0.017445      0.755792  0.754445   
ARR_TIME              0.014789 -0.000226 -0.010972      0.756969  0.757388   
ARR_DELAY            -0.095738  0.083934  0.051736      0.099208  0.155911   
ARR_DELAY_NEW        -0.056968  0.064739  0.036302      0.078027  0.136258   
CRS_ELAPSED_TIME     -0.007469  0.024239  0.015174     -0.081888 -0.075328   
ACTUAL_ELAPSED_TIME  -0.021888  0.030711  0.019712     -0.081613 -0.074904   
AIR_TIME             -0.024825  0.028719  0.018090     -0.070213 -0.063881   
DISTANCE             -0.015732  0.029055  0.017465     -0.068098 -0.061631   
CARRIER_DELAY        -0.031088  0.033890  0.018200      0.032474  0.076129   
WEATHER_DELAY        -0.000497  0.005782 -0.002961     -0.004396  0.007623   
NAS_DELAY            -0.025576  0.032416  0.017401     -0.001057  0.014639   
SECURITY_DELAY       -0.009246  0.008206  0.006006      0.004960  0.007562   
LATE_AIRCRAFT_DELAY  -0.035860  0.043357  0.026818      0.084222  0.121861   

                     DEP_DELAY  DEP_DELAY_NEW  TAXI_OUT  WHEELS_OFF  \
MONTH_DAY            -0.069803      -0.059601  0.034694   -0.006342   
WEEK_DAY              0.078215       0.068159  0.008700    0.008880   
FL_NUM                0.046779       0.039619  0.004223   -0.005186   
CRS_DEP_TIME          0.106873       0.107251 -0.107235    0.992801   
DEP_TIME              0.170378       0.169952 -0.102996    0.997369   
DEP_DELAY             1.000000       0.993313  0.038875    0.165482   
DEP_DELAY_NEW         0.993313       1.000000  0.034714    0.164870   
TAXI_OUT              0.038875       0.034714  1.000000   -0.083038   
WHEELS_OFF            0.165482       0.164870 -0.083038    1.000000   
WHEELS_ON             0.079909       0.075963 -0.066005    0.776131   
TAXI_IN               0.002023       0.001696 -0.002257   -0.017629   
CRS_ARR_TIME          0.094596       0.090775 -0.072246    0.753894   
ARR_TIME              0.076049       0.072020 -0.064825    0.759244   
ARR_DELAY             0.894162       0.888264  0.230819    0.155496   
ARR_DELAY_NEW         0.930269       0.942908  0.163794    0.133792   
CRS_ELAPSED_TIME      0.060617       0.044730  0.080117   -0.073978   
ACTUAL_ELAPSED_TIME   0.062771       0.046639  0.165157   -0.071846   
AIR_TIME              0.059572       0.043609  0.058554   -0.062965   
DISTANCE              0.061611       0.045293  0.044790   -0.061127   
CARRIER_DELAY         0.646902       0.658034  0.037770    0.073541   
WEATHER_DELAY         0.164531       0.168853  0.036969    0.008306   
NAS_DELAY             0.218601       0.218138  0.298989    0.020506   
SECURITY_DELAY        0.034918       0.034776  0.001752    0.007509   
LATE_AIRCRAFT_DELAY   0.676136       0.688995  0.034613    0.116771   

                     WHEELS_ON  ...  ARR_DELAY_NEW  CRS_ELAPSED_TIME  \
MONTH_DAY             0.013104  ...      -0.056968         -0.007469   
WEEK_DAY              0.000363  ...       0.064739          0.024239   
FL_NUM               -0.010285  ...       0.036302          0.015174   
CRS_DE

In [54]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 107708 entries, 0 to 109769
Data columns (total 30 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   MONTH_DAY            107708 non-null  int64  
 1   WEEK_DAY             107708 non-null  int64  
 2   FL_DATE              107708 non-null  object 
 3   FL_NUM               107708 non-null  int64  
 4   TAIL_NUM             107708 non-null  object 
 5   ORIGIN_CITY          107708 non-null  object 
 6   ORIGIN_STATE         107708 non-null  object 
 7   DEST_CITY            107708 non-null  object 
 8   DEST_STATE           107708 non-null  object 
 9   CRS_DEP_TIME         107708 non-null  int64  
 10  DEP_TIME             107708 non-null  float64
 11  DEP_DELAY            107708 non-null  float64
 12  DEP_DELAY_NEW        107708 non-null  float64
 13  TAXI_OUT             107708 non-null  float64
 14  WHEELS_OFF           107708 non-null  float64
 15  WHEELS_ON        

Let's convert the column FL_DATE from object to Date

In [102]:
df["FL_DATE"] = pd.to_datetime(df["FL_DATE"])

In [43]:
df.isna().any()

MONTH_DAY              False
WEEK_DAY               False
FL_DATE                False
FL_NUM                 False
TAIL_NUM               False
ORIGIN_CITY            False
ORIGIN_STATE           False
DEST_CITY              False
DEST_STATE             False
CRS_DEP_TIME           False
DEP_TIME               False
DEP_DELAY              False
DEP_DELAY_NEW          False
TAXI_OUT               False
WHEELS_OFF             False
WHEELS_ON               True
TAXI_IN                 True
CRS_ARR_TIME           False
ARR_TIME                True
ARR_DELAY               True
ARR_DELAY_NEW           True
CRS_ELAPSED_TIME       False
ACTUAL_ELAPSED_TIME     True
AIR_TIME                True
DISTANCE               False
CARRIER_DELAY          False
WEATHER_DELAY          False
NAS_DELAY              False
SECURITY_DELAY         False
LATE_AIRCRAFT_DELAY    False
dtype: bool

we need to know number of NaN in each colomn

In [53]:
df.isna().sum()  # or df.isnull().sum()

MONTH_DAY              0
WEEK_DAY               0
FL_DATE                0
FL_NUM                 0
TAIL_NUM               0
ORIGIN_CITY            0
ORIGIN_STATE           0
DEST_CITY              0
DEST_STATE             0
CRS_DEP_TIME           0
DEP_TIME               0
DEP_DELAY              0
DEP_DELAY_NEW          0
TAXI_OUT               0
WHEELS_OFF             0
WHEELS_ON              0
TAXI_IN                0
CRS_ARR_TIME           0
ARR_TIME               0
ARR_DELAY              0
ARR_DELAY_NEW          0
CRS_ELAPSED_TIME       0
ACTUAL_ELAPSED_TIME    0
AIR_TIME               0
DISTANCE               0
CARRIER_DELAY          0
WEATHER_DELAY          0
NAS_DELAY              0
SECURITY_DELAY         0
LATE_AIRCRAFT_DELAY    0
dtype: int64

In the columns CARRIER_DELAY, WEATHER_DELAY, NAS_DELAY, SECURITY_DELAY and LATE_AIRCRAFT_DELAY
The null means that there is no delay we will just replace the nulls with 0

In [39]:
#replac null with 0
df['CARRIER_DELAY'] = df['CARRIER_DELAY'].fillna(0)
df['WEATHER_DELAY'] = df['WEATHER_DELAY'].fillna(0)
df['NAS_DELAY'] = df['NAS_DELAY'].fillna(0)
df['SECURITY_DELAY'] = df['SECURITY_DELAY'].fillna(0)
df['LATE_AIRCRAFT_DELAY'] = df['LATE_AIRCRAFT_DELAY'].fillna(0)

We found nulls in columns WHEELS_ON, TAXI_IN, ARR_TIME, ACTUAL_ELAPSED_TIME and AIR_TIME

But this columns should not have nulls so we will consider it un complete data and we will exclude it from the study

In [52]:
#remove raw with null 
df = df[df.WHEELS_ON.notnull()]
df = df[df.TAXI_IN.notnull()]
df = df[df.ARR_TIME.notnull()]
df = df[df.ACTUAL_ELAPSED_TIME.notnull()]
df = df[df.AIR_TIME.notnull()]

In [8]:
df.describe() 

DAY_OF_MONTH    DAY_OF_WEEK  MKT_CARRIER_FL_NUM   CRS_DEP_TIME  \
count  109770.000000  109770.000000       109770.000000  109770.000000   
mean       15.996948       3.935602         2243.172251    1337.307780   
std         8.994348       1.908825         1705.824091     478.999221   
min         1.000000       1.000000            1.000000     500.000000   
25%         8.000000       2.000000          966.000000     915.000000   
50%        16.000000       4.000000         1744.000000    1330.000000   
75%        24.000000       5.000000         3241.750000    1740.000000   
max        31.000000       7.000000         6860.000000    2335.000000   

            DEP_TIME      DEP_DELAY  DEP_DELAY_NEW       TAXI_OUT  \
count  107879.000000  107879.000000  107879.000000  107860.000000   
mean     1341.388871       4.342968       6.615319      12.020842   
std       485.342522      21.106205      20.222221       5.975671   
min         1.000000     -20.000000       0.000000       1.000000   
25%       919.000000      -4.000000       0.000000       9.000000   
50%      1335.000000      -1.000000       0.000000      10.000000   
75%      1746.000000       4.000000       4.000000      14.000000   
max      2359.000000     602.000000     602.000000     115.000000   

          WHEELS_OFF      WHEELS_ON  ...      CANCELLED  CRS_ELAPSED_TIME  \
count  107860.000000  107814.000000  ...  109770.000000     109770.000000   
mean     1361.662618    1482.372326  ...       0.017500        128.869591   
std       484.029902     516.422930  ...       0.131126         56.278200   
min         1.000000       1.000000  ...       0.000000         35.000000   
25%       931.000000    1059.000000  ...       0.000000         85.000000   
50%      1347.000000    1515.000000  ...       0.000000        115.000000   
75%      1758.000000    1918.000000  ...       0.000000        160.000000   
max      2400.000000    2400.000000  ...       1.000000        385.000000   

       ACTUAL_ELAPSED_TIME       AIR_TIME       DISTANCE  CARRIER_DELAY  \
count        107708.000000  107708.000000  109770.000000   10321.000000   
mean            119.095796     102.059011     724.337952      18.209185   
std              55.411353      54.368042     441.776658      32.635017   
min              27.000000      18.000000      84.000000       0.000000   
25%              76.000000      60.000000     371.000000       0.000000   
50%             106.000000      89.000000     616.000000       7.000000   
75%             149.000000     131.000000     971.000000      21.000000   
max             416.000000     389.000000    2557.000000     422.000000   

       WEATHER_DELAY     NAS_DELAY  SECURITY_DELAY  LATE_AIRCRAFT_DELAY  
count   10321.000000  10321.000000    10321.000000         10321.000000  
mean        0.838485      7.873946        0.097568            18.690631  
std        10.117391     18.111303        2.034031            34.640534  
min         0.000000      0.000000        0.000000             0.000000  
25%         0.000000      0.000000        0.000000             0.000000  
50%         0.000000      0.000000        0.000000             4.000000  
75%         0.000000     10.000000        0.000000            24.000000  
max       407.000000    280.000000      117.000000           597.000000  

[8 rows x 25 columns]

To examine data types and look for cases of missing or potentially wrong data, the describe function helps to get an overview of the data from a statistical summary point of view. This is also useful for spotting any potential errors that may need a closer look.

In [44]:
df.shape

(109770, 32)

There are 109770 records and 36 columns in the dataset.

In [56]:
df.size

3231240

In [63]:
df.columns 

Index(['MONTH_DAY', 'WEEK_DAY', 'FL_DATE', 'FL_NUM', 'TAIL_NUM', 'ORIGIN_CITY',
       'ORIGIN_STATE', 'DEST_CITY', 'DEST_STATE', 'CRS_DEP_TIME', 'DEP_TIME',
       'DEP_DELAY', 'DEP_DELAY_NEW', 'TAXI_OUT', 'WHEELS_OFF', 'WHEELS_ON',
       'TAXI_IN', 'CRS_ARR_TIME', 'ARR_TIME', 'ARR_DELAY', 'ARR_DELAY_NEW',
       'CRS_ELAPSED_TIME', 'ACTUAL_ELAPSED_TIME', 'AIR_TIME', 'DISTANCE',
       'CARRIER_DELAY', 'WEATHER_DELAY', 'NAS_DELAY', 'SECURITY_DELAY',
       'LATE_AIRCRAFT_DELAY'],
      dtype='object')

In [64]:
df.dtypes # This will peovied data type for each column

MONTH_DAY                int64
WEEK_DAY                 int64
FL_DATE                 object
FL_NUM                   int64
TAIL_NUM                object
ORIGIN_CITY             object
ORIGIN_STATE            object
DEST_CITY               object
DEST_STATE              object
CRS_DEP_TIME             int64
DEP_TIME               float64
DEP_DELAY              float64
DEP_DELAY_NEW          float64
TAXI_OUT               float64
WHEELS_OFF             float64
WHEELS_ON              float64
TAXI_IN                float64
CRS_ARR_TIME             int64
ARR_TIME               float64
ARR_DELAY              float64
ARR_DELAY_NEW          float64
CRS_ELAPSED_TIME         int64
ACTUAL_ELAPSED_TIME    float64
AIR_TIME               float64
DISTANCE                 int64
CARRIER_DELAY          float64
WEATHER_DELAY          float64
NAS_DELAY              float64
SECURITY_DELAY         float64
LATE_AIRCRAFT_DELAY    float64
dtype: object

In [65]:
# i need to list all origin city name
print(df.ORIGIN_CITY.unique())

['Ontario, CA' 'Norfolk, VA' 'West Palm Beach/Palm Beach, FL'
 'Portland, OR' 'Philadelphia, PA' 'Phoenix, AZ' 'Pittsburgh, PA'
 'Pensacola, FL' 'Providence, RI' 'Portland, ME' 'Raleigh/Durham, NC'
 'Richmond, VA' 'Reno, NV' 'Rochester, NY' 'Fort Myers, FL'
 'San Diego, CA' 'San Antonio, TX' 'Louisville, KY' 'Seattle, WA'
 'San Francisco, CA' 'San Jose, CA' 'San Juan, PR' 'Salt Lake City, UT'
 'Sacramento, CA' 'Santa Ana, CA' 'St. Louis, MO' 'Tampa, FL' 'Tulsa, OK'
 'Tucson, AZ' 'Albuquerque, NM' 'Albany, NY' 'Amarillo, TX' 'Atlanta, GA'
 'Austin, TX' 'Dallas, TX' 'New Orleans, LA' 'Oakland, CA' 'Kahului, HI'
 'Oklahoma City, OK' 'Omaha, NE' 'Hartford, CT' 'Birmingham, AL'
 'Nashville, TN' 'Boise, ID' 'Boston, MA' 'Buffalo, NY' 'Burbank, CA'
 'Baltimore, MD' 'Charleston, SC' 'Cleveland, OH' 'Charlotte, NC'
 'Columbus, OH' 'Corpus Christi, TX' 'Cincinnati, OH' 'Washington, DC'
 'Denver, CO' 'Des Moines, IA' 'Detroit, MI' 'Panama City, FL'
 'El Paso, TX' 'Fort Lauderdale, FL' 'Spokane, W

In [99]:
#To check if there is duplicate rows in the dataset
sum(df.duplicated())

0

In [100]:
# This command tell us how many flights in each airport, it will basically calculate city name repeat number
print (df['ORIGIN_CITY'].value_counts())

Las Vegas, NV      5663
Dallas, TX         5635
Denver, CO         5564
Chicago, IL        5424
Baltimore, MD      5409
                   ... 
Panama City, FL      97
Pensacola, FL        91
Portland, ME         90
Des Moines, IA       85
Hilo, HI             51
Name: ORIGIN_CITY, Length: 88, dtype: int64


let's Display some Graphs

In [61]:
#We need to how many filght for each plain 
print (df.TAIL_NUM.value_counts())

N739GB    203
N942WN    202
N938WN    201
N789SW    201
N468WN    199
         ... 
N7840A     26
N767SW     19
N8312C     14
N7838A      8
N7875A      6
Name: TAIL_NUM, Length: 704, dtype: int64


In [95]:
df[df['FL_NUM'] == 795]  # This will desplay all flight number 795

MONTH_DAY  WEEK_DAY    FL_DATE  FL_NUM TAIL_NUM        ORIGIN_CITY  \
1468           30         4  1/30/2020     795   N7831B  Oklahoma City, OK   
3562           31         5  1/31/2020     795   N246LV     Washington, DC   
4019           31         5  1/31/2020     795   N246LV        Houston, TX   
5111           31         5  1/31/2020     795   N246LV  Oklahoma City, OK   
6477            6         1   1/6/2020     795   N779SW     Washington, DC   
6931            6         1   1/6/2020     795   N779SW        Houston, TX   
8016            6         1   1/6/2020     795   N779SW  Oklahoma City, OK   
10123           7         2   1/7/2020     795   N797MX     Washington, DC   
10580           7         2   1/7/2020     795   N797MX        Houston, TX   
11671           7         2   1/7/2020     795   N797MX  Oklahoma City, OK   
13009          13         1  1/13/2020     795   N7831B        Houston, TX   
13468          16         4  1/16/2020     795   N7861J     Washington, DC   
13924          16         4  1/16/2020     795   N7861J        Houston, TX   
15195          27         1  1/27/2020     795   N707SA        Houston, TX   
16291          27         1  1/27/2020     795   N707SA  Oklahoma City, OK   
18388          28         2  1/28/2020     795   N292WN     Washington, DC   
18844          28         2  1/28/2020     795   N292WN        Houston, TX   
21064           3         5   1/3/2020     795   N969WN     Fort Myers, FL   
23296           9         4   1/9/2020     795   N7744A  Oklahoma City, OK   
25395          10         5  1/10/2020     795   N712SW     Washington, DC   
25849          10         5  1/10/2020     795   N712SW        Houston, TX   
26938          10         5  1/10/2020     795   N712SW  Oklahoma City, OK   
29355          16         4  1/16/2020     795   N7861J  Oklahoma City, OK   
31456          17         5  1/17/2020     795   N268WN     Washington, DC   
31913          17         5  1/17/2020     795   N268WN        Houston, TX   
33074          30         4  1/30/2020     795   N7831B     Washington, DC   
34837           9         4   1/9/2020     795   N7744A     Washington, DC   
35294           9         4   1/9/2020     795   N7744A        Houston, TX   
37255          15         3  1/15/2020     795   N7851A     Washington, DC   
37711          15         3  1/15/2020     795   N7851A        Houston, TX   
38802          15         3  1/15/2020     795   N7851A  Oklahoma City, OK   
42408          14         2  1/14/2020     795   N964WN     Washington, DC   
42860          14         2  1/14/2020     795   N964WN        Houston, TX   
43949          14         2  1/14/2020     795   N964WN  Oklahoma City, OK   
44845          20         1  1/20/2020     795   N556WN     Washington, DC   
45301          20         1  1/20/2020     795   N556WN        Houston, TX   
46397          20         1  1/20/2020     795   N556WN  Oklahoma City, OK   
50188          27         1  1/27/2020     795   N707SA     Washington, DC   
50695           2         4   1/2/2020     795   N245WN     Fort Myers, FL   
68897          23         4  1/23/2020     795   N220WN  Oklahoma City, OK   
70995          24         5  1/24/2020     795   N958WN     Washington, DC   
72548          24         5  1/24/2020     795   N958WN  Oklahoma City, OK   
75974          29         3  1/29/2020     795   N7735A     Washington, DC   
76430          29         3  1/29/2020     795   N7735A        Houston, TX   
77527          29         3  1/29/2020     795   N7735A  Oklahoma City, OK   
82402           8         3   1/8/2020     795   N7750A     Washington, DC   
82859           8         3   1/8/2020     795   N7750A        Houston, TX   
83950           8         3   1/8/2020     795   N7750A  Oklahoma City, OK   
85476          28         2  1/28/2020     795   N292WN  Oklahoma City, OK   
91798          22         3  1/22/2020     795   N497WN  Oklahoma City, OK   
93892          23         4  1